datafusion/sqllogictest/test_files/array.slt

```diff
@@ -5549,26 +5549,32 @@ from arrays_range;
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] [3, 4, 5, 6, 7, 8, 9] [3, 5, 7, 9]
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] [4, 5, 6, 7, 8, 9, 10, 11, 12] [4, 7, 10]
 
-query ??????
+query ?????????
 select range(5),
        range(2, 5),
        range(2, 10, 3),
        range(1, 5, -1),
        range(1, -5, 1),
-       range(1, -5, -1)
+       range(1, -5, -1),
+       range(DATE '1992-09-01', DATE '1993-03-01', INTERVAL '1' MONTH),
+       range(DATE '1993-02-01', DATE '1993-01-01', INTERVAL '-1' DAY),
+       range(DATE '1989-04-01', DATE '1993-03-01', INTERVAL '1' YEAR)
```

Comment:
Can you please also add tests for null handling?

For example

```sql
       range(DATE '1992-09-01', DATE '1993-03-01', NULL,
       range(DATE '1992-09-01', NULL , INTERVAL '1' MONTH),
       range(NULL, DATE '1993-03-01', INTERVAL '1' MONTH),
```

Also, can you add negative tests (like when start is > stop by the the interval is positive (I think the code will error in this case, but it would be good to verify this

```sql
       range(DATE '1993-03-01', DATE '1989-04-01', INTERVAL '1' YEAR)
       range(, DATE '1989-04-01', DATE '1993-03-01', INTERVAL '-1' YEAR)
```

Reply:
OK, I will do it now